### PR TITLE
[Docs] Build and Release: Update ci-build-and-release.md

### PR DESCRIPTION
**Pull request by leecalcote**
**Notes for Reviewers**

- This PR enhances the build and release docs page with quick references to important documents.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by theBeginner86 — APPROVED** [review]

**Review by lekaf974 — CHANGES_REQUESTED** [review]
broke the tests

**Comment by leecalcote** [comment]
Thank you, @lekaf974. A pair of double quotes was missing.

**Review by ritzorama — APPROVED** [review]

**Review by marblom007 — DISMISSED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit a02805a3e1f6b6e66bc705f67900f2f23f8c1c8c at: https://6747bf8f8eebc0d423238828--meshery-docs-preview.netlify.app

File: docs/pages/project/contributing/ci-build-and-release.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ list: include
...
+{% include alert.html type="info" title="Meshery Test Documents" content="The <a href='https://docs.google.com/spreadsheets/d/13Ir4gfaKoAX9r8qYjAFFl_U9ntke4X5ndREY1T7bnVs/edit'>Meshery Test Plan</a> is a tactical spreadsheet that categories specific integration and end-to-end test cases, itemizing each test case and their status. It includes a list of GitHub workflows and their purpose. The Meshery Test Strategy (<a href='https://docs.google.com/document/d/11nAxYtz2SUusCYZ0JeNRrOLIxkgmmbUVWz63MBZV2oE/edit'>1</a>,<a href='https://docs.google.com/document/d/14vbwnKafqxrr-cJOmLWEvvj75MFrxFKiUdfPwZRhg64/edit'>2</a>) is a comprehensive document that outlines the approach to testing, types of tests, frameworks used - test strategy - for each of Meshery's architectural components. Join the <a href='https://docs.google.com/document/d/1GrVdGHZAYeu6wHNLLoiaKNqBtk7enXE9XeDRCvdA4bY/edit#'>Meshery CI meeting</a> and/or the <a href='https://meshery.io/subscribe'>developers mailing list</a> to get involved." %}
```
Comment (ianrwhitney) [inline docs/pages/project/contributing/ci-build-and-release.md:14]: [nit] -`Meshery Test Plan is a tactical spreadheet that **categorizes** specific integratione ...`

**Review by ianrwhitney — COMMENTED** [review]
LGTM one nit for your consideration. thanks

File: docs/pages/project/contributing/ci-build-and-release.md (hunk 1 above)
Comment (leecalcote) [reply docs/pages/project/contributing/ci-build-and-release.md:14]: Very good. Thank you, @ianrwhitney

Comment (leecalcote) [inline docs/pages/project/contributing/ci-build-and-release.md:14]:
```suggestion
{% include alert.html type="info" title="Meshery Test Documents" content="The <a href='https://docs.google.com/spreadsheets/d/13Ir4gfaKoAX9r8qYjAFFl_U9ntke4X5ndREY1T7bnVs/edit'>Meshery Test Plan</a> is a tactical spreadsheet that **categories** specific integration and end-to-end test cases, itemizing each test case and their status. It includes a list of GitHub workflows and their purpose. The Meshery Test Strategy (<a href='https://docs.google.com/document/d/11nAxYtz2SUusCYZ0JeNRrOLIxkgmmbUVWz63MBZV2oE/edit'>1</a>,<a href='https://docs.google.com/document/d/14vbwnKafqxrr-cJOmLWEvvj75MFrxFKiUdfPwZRhg64/edit'>2</a>) is a comprehensive document that outlines the approach to testing, types of tests, frameworks used - test strategy - for each of Meshery's architectural components. Join the <a href='https://docs.google.com/document/d/1GrVdGHZAYeu6wHNLLoiaKNqBtk7enXE9XeDRCvdA4bY/edit#'>Meshery CI meeting</a> and/or the <a href='https://meshery.io/subscribe'>developers mailing list</a> to get involved." %}
```

**Pull request merged**